src/backend/function/Fusion.Resources.Functions/Functions/Notifications/ScheduledReportTimerTriggerFunction.cs

```diff
@@ -69,31 +69,38 @@ private async Task SendResourceOwnersToQueue(ServiceBusSender sender)
             var selectedDepartments = departments
                 .Where(d => d.FullDepartment != null && d.FullDepartment.Contains("PRD")).Distinct().ToList();
             var resourceOwners = await GetLineOrgPersonsFromDepartmetnsChunked(selectedDepartments);
-
             if (resourceOwners == null || !resourceOwners.Any())
                 throw new Exception("No resource-owners found.");
 
-            foreach (var resourceOwner in resourceOwners.DistinctBy(ro => ro.AzureUniqueId))
+            var resourceOwnersToSendNotifications = resourceOwners.DistinctBy(ro => ro.AzureUniqueId);
+
+            var batchTime = Math.Ceiling(60.0 / resourceOwnersToSendNotifications.Count());
```

Comment:
If we start the function earlier we can send the notifications out over two or three hours instead of one.